kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala

```diff
@@ -419,6 +419,142 @@ class JDBCMetadataStore(conf: KyuubiConf) extends MetadataStore with Logging {
     }
   }
 
+  private def insertKubernetesEngineInfo(engineInfo: KubernetesEngineInfo): Unit = {
+    val insertQuery =
+      s"""
+         |INSERT INTO $KUBERNETES_ENGINE_INFO_TABLE(
+         |identifier,
+         |context,
+         |namespace,
+         |pod_name,
+         |pod_state,
+         |container_state,
+         |engine_id,
+         |engine_name,
+         |engine_state,
+         |engine_error,
+         |create_time,
+         |update_time
+         |)
+         |SELECT ?, ?, ?, ?, ?, ?, ?, ?, ?, ?,?,?
+         |WHERE NOT EXISTS (
```

Comment:
not sure if it's better to use the DB-specific dialect, e.g. MySQL supports `INSERT IGNORE INTO`